packages/realm/binding/android/build.gradle

```diff
@@ -88,6 +95,16 @@ android {
         }
     }
 
+    sourceSets {
+        main {
+            if (NEW_ARCH_ENABLED) {
+                java.srcDirs += ["src/newarch"]
+            } else {
+                java.srcDirs += ["src/oldarch"]
+            }
+        }
+    }
```

Comment:
This is the magic that controls which implementation is used based on the new arch being enabled for Android.

Reply:
🪄